pkg/controller/elasticsearch/driver/nodes.go

```diff
@@ -212,13 +214,25 @@ func (d *defaultDriver) reconcileNodeSpecs(
 		return results
 	}
 
+	isNodeSpecsReconciled := d.isNodeSpecsReconciled(actualStatefulSets, d.Client, results)
 	// as of 7.15.2 with node shutdown we do not need transient settings anymore and in fact want to remove any left-overs.
-	if esReachable && d.isNodeSpecsReconciled(actualStatefulSets, d.Client, results) {
+	if esReachable && isNodeSpecsReconciled {
 		if err := d.maybeRemoveTransientSettings(ctx, esClient); err != nil {
 			return results.WithError(err)
 		}
 	}
 
+	// Set or update an orchestration hint to let the association controller know of service account are supported.
+	if isNodeSpecsReconciled {
```

Comment:
This looks out of place are we not setting the orchestration hint in `driver.go`?

Reply:
I think this whole block should go away. It is true that `isNodeSpecReconciled` is kind of a proxy for "all nodes should have the service account token now" but it is not as accurate as observing it through the `_security` API which is what we are doing anyway.

Reply:
I think finally start to grasp your idea. You are waiting for the symlink to be present even before rolling out any service account tokens!